identity-apps-core/apps/recovery-portal/src/main/webapp/recovery.jsp

```diff
@@ -65,12 +75,7 @@
     }
 
     if (StringUtils.isNotBlank(username)) {
-        if (StringUtils.isNotBlank(userTenantHint)) {
-            username = MultitenantUtils.getTenantAwareUsername(username);
-            username = UserCoreUtil.addTenantDomainToEntry(username, userTenantHint);
-        } else {
-            username = UserCoreUtil.addTenantDomainToEntry(username, userTenant);
-        }
+        username = (String) request.getAttribute("resolvedUsername");
```

Comment:
Check whether we need to sanitize this with Encode.forJava as well

Reply:
Don't need to sanitize this attribute since it's getting set in the username-resolver.jsp file